### PR TITLE
[query] Fix bad tuple indexing on PCanonicalBaseStructSettable

### PR DESCRIPTION
**Pull request by johnc1231**
This should fix Julia's bug here: https://discuss.hail.is/t/arrayindexoutofboundsexception-error-in-hail-0-2-40/1413/11

I will make a follow up PR tomorrow splitting things up and making a `PCanonicalTupleCode`. I just wanted to get rid of this blatantly wrong thing.

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/types/physical/PCanonicalBaseStruct.scala [hunk 1]
```diff
@@ -242,23 +242,30 @@ class PCanonicalBaseStructSettable(
 
   def settableTuple(): IndexedSeq[Settable[_]] = FastIndexedSeq(a)
 
-  def loadField(cb: EmitCodeBuilder, fieldIdx: Int): IEmitCode = {
+  def loadField(cb: EmitCodeBuilder, i: Int): IEmitCode = {
+    val ptTuple = pt.asInstanceOf[PTuple]
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/expr/types/physical/PCanonicalBaseStruct.scala:246]: I think this is the wrong fix. PCBaseStructSettable doesn't seem broken -- the fieldIdx here should just be base struct index, not tuple index.

The fix should be to use the true index here: https://github.com/hail-is/hail/blob/50d0da730381e3afbb6a6ea26ebf705597f5549d/hail/src/main/scala/is/hail/expr/ir/Emit.scala#L730

Looks like Chris's PR that moved a bunch of Emit to IEmit broke this on 4/22.

Comment (johnc1231) [reply hail/src/main/scala/is/hail/expr/types/physical/PCanonicalBaseStruct.scala:246]: Ah, ok that's a better idea. One sec

**Comment by johnc1231** [comment]
Fixed! Thanks.

**Comment by johnc1231** [comment]
Added WIP because about to push fixes that still have some debugging info floating around

**Pull request merged**